3.workload.md

```diff
@@ -0,0 +1,128 @@
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `name` | `string` | Y | | The workload schema's name. |
+
+The workload schema definition itself __MUST__ contain information that can be used to uniquely identify it. An OAM runtime will use that information in a [Component](4.component.md) to refer to the workload schema instead of using the name of the workload definition.
+ We recommend to use the format [Group/Version/Kind](2.overview_and_terminology.md#Group) described in section 2 as the unique identification in the schema. 
+
+We recommend that the `name` here follows the format described in [the Representations of Group/Kind in section 2](2.overview_and_terminology.md#Representations).
+
+We also recommend that the `name` of the workloadDefinition to be the same as the `name` it refers to. Here is an example
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: WorkloadDefinition
+metadata:
+  name: schema.example.oam.dev
+spec:
+  definitionRef:
+    schemaName: schema.example.oam.dev
+```
+
+## Categories of Workload
+
+There are three categories of workloads:
+
+__Core workloads__ are the kind of workloads that every OAM runtime MUST implement according to spec definition.
+
+__Standard workloads__ are the kind of workloads that a OAM runtime can choose to implement. However, the implementation MUST strictly adhere to the schematics defined in the spec. 
+
+__Extended workloads__ are the kind of workloads that a OAM runtime is free to define.
+
+### Core Workload
+
+A core workload schema MUST be in the `core.oam.dev` group. A core workload MUST all be supported according to the definitions in the spec by any runtime implementation of this specification. Here is an example of a core workload definition.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: WorkloadDefinition
+metadata:
+  name: x.core.oam.dev
+spec:
+  definitionRef:
+    name: x.core.oam.dev
+```
+
+Currently, all core workloads are container-based, and assume that an implementation is capable of executing an OCI or Docker image as a container, and are capable of working with OCI registries.
+We may introduce non container-based workloads, such as VM-based, in the future. We can also promote standard workloads to core if there is a consensus in the community.
+
+### Standard Workload
+
+A standard workload schema MUST be in the `standard.oam.dev` group. Here is an example of a standard workload definition.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: WorkloadDefinition
+metadata:
+  name: x.standard.oam.dev
+spec:
+  definitionRef:
+    name: x.standard.oam.dev
+```
+
+A standard workload MUST be implemented according to the definitions in the spec if any runtime chooses to implement it.
+ 
+We do not have any standard workload defined in the current spec. We may introduce new workloads or promote extended workloads to standard if there is a consensus in the community.
```

Comment:
Can we make a clear claim that standard workloads will be promoted from extended workloads? I'm hoping users can get the point of "promotion ladder" in this section.

Reply:
I am not sure if the standard workloads have to be promoted from an extended one. It can be a new one that is similar to a bunch of extended workloads. The core one has to be promoted from standard as we won't change its definition.

Reply:
Emm, can we highlight the idea of promotion ladder somewhere? It seems implicit for now.